src/locutus/model/terminology.py

```diff
@@ -436,18 +479,51 @@ def add_provenance(
         self, change_type, editor, target=None, timestamp=None, **kwargs
     ):
         if target is None:
-            target = "self"
+            code_index = "self"
+            normalized_target = "self"
+        else:
+            # Ensure special characters in paired targets are handled properly
+            if "|" in target:
+                left, right = target.split("|", 1)
+                # Normalize each side of a paired target
+                normalized_left = (
+                    normalize_ftd_placeholders(left)
+                    if left in FTD_PLACEHOLDERS
+                    else left
+                )
+                normalized_right = (
+                    normalize_ftd_placeholders(right)
+                    if right in FTD_PLACEHOLDERS
+                    else right
+                )
+                normalized_target = generate_paired_string(
+                    normalized_left, normalized_right
+                )
+                # Ensure indexes are formatted properly
+                index_left = get_code_index(left) if left in FTD_PLACEHOLDERS else left
+
+                index_right = (
+                    get_code_index(right) if left in FTD_PLACEHOLDERS else right
+                )
+
+                index_right = get_code_index(right)
+                code_index = generate_paired_string(index_left, index_right)
+            # Ensure special characters in single code targets are handled properly
+            else:
+                normalized_target = normalize_ftd_placeholders(target)
+                code_index = get_code_index(target)
+
         if timestamp is None:
             timestamp = datetime.now()
 
         timestamp = timestamp.strftime(PROVENANCE_TIMESTAMP_FORMAT)
         # cur_prov = None
-        cur_prov = self.get_provenance(target)[target]
+        cur_prov = self.get_provenance(code_index).get(normalized_target, None)

```

Comment:
This is a "gut feeling", and not a hard suggestion, but I suspect it may be better to move those code indexes into the get_provenance and get_mapping, etc functions. That won't be correct if there places you have to use the code_index directly. But, similarly to earlier comments, keeping the purpose close to the need can shorten logic updates if there are changes that must be made down the line.